app/controllers/todos_controller.rb

```diff
@@ -1,17 +1,24 @@
 class TodosController < ApplicationController
+
   before_action :authenticate_user!   # Ensure the user is authenticated via token
   before_action :set_todo_list
-  before_action :set_todo, only: [:toggle_status, :update, :destroy]
+  before_action :set_todo, only: [:toggle_status, :update, :destroy, :create]
 
   # GET /users/:user_id/todo_lists/:todo_list_id/todos
   def index
-    @todos = @todo_list.todos
-    render json: @todos
+    @todo = policy_scope(Todo)
+    if @todo.none?
+      render json: { error: 'Not Found' }, status: :not_found
+    else
+      render json: @todo
+    end  
+  
   end
 
   # POST /users/:user_id/todo_lists/:todo_list_id/todos
   def create
     @todo = @todo_list.todos.build(todo_params)
+    authorize @todo
```

Comment:
authorize after build?

Reply:
Basically authorize method requires a object to exist for evaluate its attributes.